workflow-ui/core-common/src/main/java/com/squareup/workflow1/visual/Wrapper.kt

```diff
@@ -0,0 +1,69 @@
+package com.squareup.workflow1.visual
+
+import com.squareup.workflow1.ui.Compatible
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * Base class for wrapper rendering types. Ensures that [Compatible] is implemented
+ * correctly, and a more regular structure should be an aid to test code if nothing
+ * else.
+ */
+@WorkflowUiExperimentalApi
+public abstract class Wrapper<W : Any>(
+  public val wrapped: W,
+) : Compatible {
+  /** Read only once by [compatibilityKey], at construction time. */
+  protected open val name: String = this::class.qualifiedName ?: "Wrapper"
+
+  final override val compatibilityKey: String by lazy { Compatible.keyFor(wrapped, name) }
```

Comment:
May be we should also add a qualified into the compatibility key?
That qualifier might be `""` by default but overridable?
For instance: WithName.qualifier should be name:
Named(slot1)-content should not be compatible with Named(slot2)-content.

Reply:
What would that provided that the name argument to `Compatible.keyFor()` doesn't provide already?